### ISSUE TITLE
Vagrant Will Not Override Cached Metadata URL

### ISSUE THREAD

**Issue by chadrobertson75** [hashicorp/vagrant#4235]
When updating the vm.box_url configuration Vagrant does not use the configuration URL if there is an existing cached box. The only way to get Vagrant to use the new URL configuration is either manually updating metadata_url in the cached box directory or deleting the cache entirely. The only automated alternative is changing the box name.

It seems that Vagrant should obey the configuration regardless of the cached location so users can possibly change the location of versioning without needing to instruct users to manually muck around in the cache directories or without changing the box name.

**Comment by ccnokes** [comment]
+1 for chadrobertson75's comments. I just spent half a day trying to figure out why changes to the box_url weren't taking effect when I ran `vagrant reload`. The way it's "working as intended" isn't very intuitive.

**Comment by chadrobertson75** [comment]
There are only two hard things in Computer Science: cache invalidation and naming things and off by one errors.

**Comment by chadrobertson75** [comment]
Again I have ran into this problem. We changed our internally hosted domain and now every developer using a vagrant box gets errors when they up a vagrant box because the old address doesn't exist in our network. Even though the Vagrantfile has the correct URL. I don't understand why the cache is the point of authority instead of the configuration. There needs to be a way to invalidate that cache beyond deleting every box from the developers machine and starting with nothing. @mitchellh I would hope this makes sense to you as a completely unexpected behavior. Considering the metadata_url cache is hidden from the user. This continues to be a pain point that is unnecessary.
